### PR TITLE
Updates to include better config and bootstrapping

### PR DESCRIPTION
**Pull request by ttschampel**
1. Add bootstrapping
2. Pull in OTDF charts directly to avoid nested chart templating issues

### CODE REVIEW

File: scp/templates/bootstrap/bootstrap-job-secret.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+{{- if .Values.secrets.enabled }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "common.lib.fullname" . }}-keycloakbootstrap-secret
+  labels:
+    {{- include "common.lib.labels" . | nindent 4 }}
+type: Opaque
+stringData:
+  CLIENT_SECRET: {{ .Values.secrets.keycloakBootstrap.clientSecret }}
```
Comment (stephaniegraham) [inline scp/templates/bootstrap/bootstrap-job-secret.yaml:14]: There is only one client secret here, are we only installing one client?  I thought we were doing one client per PEP?

File: scp/values.yaml [hunk 2]
```diff
@@ -2,10 +2,47 @@
...
 fullnameOverride: ""
 
+embedded:
+  # -- Use an embedded keycloak or not
+  keycloak: true
+  # -- Use an embedded postgres or not
+  postgresql: true
+
+bootstrap:
+  enabled: false
```
Comment (stephaniegraham) [inline scp/values.yaml:27]: what does this overall enabled mean?

File: scp/values.yaml [hunk 3]
```diff
@@ -31,59 +77,147 @@ abacus:
...
+    postgres:
+      host: postgresql
+      # Override with value
+      password:
+    imagePullSecrets: *imagePullSecrets
+
+# Entitlement Policy Bootstrap
+entitlement-policy-bootstrap:
+  policyGlobPattern: "configs/fed-demo/entitlement-policy/*"
+  bundleRepo: *entitlementPolicyBundleRepo
```
Comment (stephaniegraham) [inline scp/values.yaml:115]: Richard is creating a new dedicated entitlement policy repo for fed.  Is this the one you will use?  https://github.com/virtru-corp/entitlement-policy-federal

**Review by stephaniegraham — COMMENTED** [review]
So, i am not the best one to review helm charts, but from a concept perspective, here are my comments.

Currently, we have manual steps for the following things that it wasn't 100% clear if they are included below (or needed anymore).

* Reverse Proxy and Firewall  (not sure if this is needed for Istio deployment, and maybe needs to be handled in the SCP charts and not here? (https://virtru.atlassian.net/wiki/spaces/FED/pages/2883911838/Customer+Hosted+SC+Platform+Configure+the+Reverse+Proxy+and+Firewall)
* Setting up Abacus (<-- I think this is included, but I didn't see the actual values being used, I would be happy to review to make sure users and abacus are inline)
* Keycloak clients (again, I think this is being included, but not sure if there is just one?  https://virtru.atlassian.net/wiki/spaces/FED/pages/2883879077/Customer+Hosted+SC+Platform+Configure+Keycloak+Clients)

File: scp/values.yaml (hunk 3 above)
Comment (ttschampel) [reply scp/values.yaml:115]: That is where I sourced my current install config from. The way this chart is currently setup is those files sit in some place relative to the installation dir (in this case /configs/fed-demo/entitlement-policy ) and they are picked up by the chart to bundle and push to the repo

File: scp/templates/bootstrap/bootstrap-job-secret.yaml (hunk 1 above)
Comment (ttschampel) [reply scp/templates/bootstrap/bootstrap-job-secret.yaml:14]: This setting here is only for the hard-wired clients the bootstrapper creates now in OTDF.

All of the custom clients can be configured via the clients list ref'd [here](https://github.com/virtru-corp/virtru-charts/blob/add-config-to-helm-repo_bootstrapping/scp/templates/bootstrap/bootstrap-config-secret.yaml#L18): 

Some brief description in this is [here](https://github.com/virtru-corp/virtru-charts/blob/add-config-to-helm-repo_bootstrapping/scp/README.md#bootstrapping-deployment):

**Comment by ttschampel** [comment]
> So, i am not the best one to review helm charts, but from a concept perspective, here are my comments.
> 
> Currently, we have manual steps for the following things that it wasn't 100% clear if they are included below (or needed anymore).
> 
> * Reverse Proxy and Firewall  (not sure if this is needed for Istio deployment, and maybe needs to be handled in the SCP charts and not here? (https://virtru.atlassian.net/wiki/spaces/FED/pages/2883911838/Customer+Hosted+SC+Platform+Configure+the+Reverse+Proxy+and+Firewall)
> * Setting up Abacus (<-- I think this is included, but I didn't see the actual values being used, I would be happy to review to make sure users and abacus are inline)
> * Keycloak clients (again, I think this is being included, but not sure if there is just one?  https://virtru.atlassian.net/wiki/spaces/FED/pages/2883879077/Customer+Hosted+SC+Platform+Configure+Keycloak+Clients)


Reverse Proxy and Firewall : Yes, no longer required with the addition of istio
Setting up Abacus:  Yes, this is done via the abacus chart inclusion + abacus value overrides + istio ingress templates

File: scp/Chart.yaml [hunk 4]
```diff
@@ -1,20 +1,58 @@
...
-dependencies:
-  - name: backend
-    repository: oci://ghcr.io/opentdf/charts
-    version: '0.0.0-sha-03c7ccc'
-    condition: backend.enabled
+otdfVersion: &otdfVersion "0.0.0-sha-bd7a8d2"
+otdfRepo: &otdfRepo "oci://ghcr.io/opentdf/charts"
 
+dependencies:
   - name: abacus
```
Comment (pflynn-virtru) [inline scp/Chart.yaml:12]: no abacus in self-hosted

File: index.yaml [hunk 5]
```diff
@@ -94,9 +94,87 @@ entries:
...
+    - name: entity-resolution
+      repository: oci://ghcr.io/opentdf/charts
+      version: 0.0.0-sha-da7b457
+    - name: kas
+      repository: oci://ghcr.io/opentdf/charts
+      version: 0.0.0-sha-da7b457
+    - name: common-lib
+      repository: file://../common-lib
+      version: 0.1.0
+    - condition: embedded.postgresql
```
Comment (pflynn-virtru) [inline index.yaml:126]: embedded postgresql is not supported for production environments.

**Comment by elizabethhealy** [comment]
what were the templating issues you were running into as a result of nesting?

File: scp/Chart.yaml (hunk 4 above)
Comment (ttschampel) [reply scp/Chart.yaml:12]: Near term we need abacus AFAIK,  you can current disable with: abacus.enabled

File: index.yaml (hunk 5 above)
Comment (ttschampel) [reply index.yaml:126]: If you want to use external postgres you can set embedded.postgresql to false

**Review by pflynn-virtru — APPROVED** [review]

**Pull request merged**